### PR TITLE
Add Explore sample app

### PR DESCRIPTION
**Pull request by imraymondlee**
Adding a sample app that provides similar maps, places, and routes functionalities to the Explore app in the Amazon Location Service Console.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by dreamorosi** [comment]
Hi @imraymondlee thanks for the new sample, looking forward to reviewing this early next week (likely on Monday 2nd).

Question: would you be open on including an amplify folder to create & deploy all the required resources with `amplify init` & `amplify deploy`? This would be in addition to the manual instructions that you have already included, it would just provide a quick alternative for those users who are into Amplify CLI.

If you like the idea you could reuse the `amplify` directory from [this other sample](https://github.com/imraymondlee/amazon-location-samples/tree/main/react-map-gl-amplify-here-map/amplify). If instead this is part of a different content strategy and the instructions must stay as they are disregard the question.

**Comment by imraymondlee** [comment]
That's a good suggestion, I will need to discuss with the team about that.

**Comment by imraymondlee** [comment]
Spoke with the team, and we have decided that we will not be adding the Amplify CLI support at the moment. I have also received some additional feedback and pushed a commit for those changes.

File: amplify-ui-geo-explore/src/App.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  const handleViewportChangeFromLayer = (bbox) => {
+    if (bbox) {
+      // Adjusting viewport based on bounding box array input
+      const boundingBox = [
+        [bbox[0], bbox[1]],
+        [bbox[2], bbox[3]],
+      ];
+
+      // Get the new viewport based on the bounding box
+      const newViewport = new WebMercatorViewport({
```
[the remark this answers is not in the document]
Comment (dreamorosi) [reply amplify-ui-geo-explore/src/App.js:84]: This means you can remove the explicit dependency to `@math.gl/web-mercator`

File: amplify-ui-geo-explore/src/App.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+import RoutesLayer from "./components/routes/RoutesLayer";
+import { MapView } from "@aws-amplify/ui-react";
+import { NavigationControl } from "react-map-gl";
+import WebMercatorViewport from "@math.gl/web-mercator";
+import "@aws-amplify/ui-react/styles.css";
+import "./index.css";
+
+// Configuring Amplify Geo with existing Amazon Cognito and Amazon Location Service information
+// Fill values in configuration.js
+Amplify.configure({
```
Comment (dreamorosi) [inline amplify-ui-geo-explore/src/App.js:18]: This works and the demo works fine, but I get errors in the console (see screenshot below):

Screenshot 2022-05-02 at 10 29 16

Can we look into it and see if we can remove them?

Comment (dreamorosi) [reply amplify-ui-geo-explore/src/App.js:18]: [Maybe helpful](https://docs.amplify.aws/lib/client-configuration/configuring-amplify-categories/q/platform/js/#general-configuration).

File: amplify-ui-geo-explore/src/components/places/PlacesLayer.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      position="top-left"
+      placeholder="Search Places"
+      showResultsWhileTyping={false}
+      showResultMarkers={markerIcon}
+      popup={popup}
+      limit={MAX_RESULT}
+      popupRender={popupRender}
+      render={render}
+      marker={markerIcon}
+    />
```
Comment (dreamorosi) [inline amplify-ui-geo-explore/src/components/places/PlacesLayer.js:96]: At least when I run it I don't see the "routing" icon in the button like the screenshot attached to this repo shows: 
image

File: amplify-ui-geo-explore/src/components/places/PlacesLayer.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  );
+};
+
+// Layer in the app that contains Places functionalities
+const PlacesLayer = () => {
+  return (
+    <Geocoder
+      position="top-left"
+      placeholder="Search Places"
+      showResultsWhileTyping={false}
```
Comment (dreamorosi) [inline amplify-ui-geo-explore/src/components/places/PlacesLayer.js:89]: Why is this set to `false`? While using it I was a bit confused as I didn't know I had to press "enter" and waited a few seconds, and nothing happened.

**Comment by dreamorosi** [comment]
Overall the demo looks good.

I found the Place Index & Routing features a bit disconnected though. For instance if I search a place using the geocoder component, and then click "routes" I would kind of expect the marker I just placed to be used in the route. Instead what happens is that the black marker stays there, and I have to select again start & destination points for my route:
image


If the above is too much work, at least make the black mark disappear & reset the geocoder state so that you don't have two completely separate things on the map I'm looking at.

**Review by dreamorosi — CHANGES_REQUESTED** [review]

File: amplify-ui-geo-explore/src/App.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (imraymondlee) [reply amplify-ui-geo-explore/src/App.js:84]: Unfortunately, Amplify UI's `MapView` component doesn't expose `ref` that is required before being able to make calls to `fitBounds` and `flyTo`. We will have to continue with the use of WebMercatorViewport until they allow the use of `ref` implemented. https://github.com/aws-amplify/amplify-ui/issues/1790

File: amplify-ui-geo-explore/src/components/places/PlacesLayer.js (hunk 3 above)
Comment (imraymondlee) [reply amplify-ui-geo-explore/src/components/places/PlacesLayer.js:96]: Fixed

File: amplify-ui-geo-explore/src/App.js (hunk 2 above)
Comment (imraymondlee) [reply amplify-ui-geo-explore/src/App.js:18]: This is an existing issue with Amplify Auth https://github.com/aws-amplify/amplify-js/issues/9845. I have left notes in the README of the sample app describing this issue and will remove it when Amplify has the fix ready.

**Pull request merged**